execnet/gateway.py

```diff
@@ -75,7 +75,10 @@ def _rinfo(self, update=False):
         """ return some sys/env information from remote. """
         if update or not hasattr(self, "_cache_rinfo"):
             ch = self.remote_exec(rinfo_source)
-            self._cache_rinfo = RInfo(ch.receive())
+            try:
+                self._cache_rinfo = RInfo(ch.receive())
+            finally:
+                ch.waitclose()
```

Comment:
Since the waitclose doesn't elicit any response to guarantee that the main thread has been released, I added a one second timeout for it to release as shown in https://github.com/pytest-dev/execnet/pull/243#discussion_r1495103598.